integration/webhook_test.go

```diff
@@ -0,0 +1,175 @@
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	k8sApiErrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+	"github.com/openshift/addon-operator/integration"
+)
+
+const (
+	CATALOG_SOURCE_URL = "quay.io/osd-addons/reference-addon-index@sha256:58cb1c4478a150dc44e6c179d709726516d84db46e4e130a5227d8b76456b5bd"
```

Comment:
A few catalog source images are pre-defined in fixtures_test.go.
This image is already defined there as `referenceAddonCatalogSourceImageWorking`.

We have to be a bit careful that we don't reference a bunch of random image sha's, or we will have no clue what those images actually contain.